alacritty/src/window_context.rs

```diff
@@ -415,7 +420,21 @@ impl WindowContext {
         event: WinitEvent<Event>,
     ) {
         match event {
-            WinitEvent::AboutToWait | WinitEvent::RedrawRequested(_) => {
+            WinitEvent::AboutToWait => {
+                // Skip further event handling with no staged updates.
+                if self.event_queue.is_empty() {
+                    // Request a redraw in-case we have dirty flag set without events in the queue.
+                    // This could happen when event excluded from batching set `dirty` flag, but
+                    // forgot to call `redraw_requested`.
```

Comment:
That's the handling for the missed `redraw_requested` I had a `NOTE` before. I don't really like it, but it protects against not asking for redraw at all.

The issue is because when we handle events without batching, we'll have an `empty` queue here, which means that we won't ask for the redraw event at all.